python/pyarrow/orc.py

```diff
@@ -22,6 +22,7 @@
 from pyarrow.lib import Table
 import pyarrow._orc as _orc
 from pyarrow.fs import _resolve_filesystem_and_path
+from pyarrow import filesystem as legacyfs
```

Comment:
Why are you importing the old filesystem API here?

Reply:
Sorry I really just copied some stuff from `parquet.py` which turned out to be irrelevant to the BytesIO bug.